crates/storage/provider/src/providers/mod.rs

```diff
@@ -181,6 +181,50 @@ impl<DB: Database> TransactionsProvider for ShareableDatabase<DB> {
             .map_err(Into::into)
     }
 
+    fn transaction_by_hash_with_meta(
+        &self,
+        tx_hash: TxHash,
+    ) -> Result<Option<(TransactionSigned, TransactionMeta)>> {
+        self.db
+            .view(|tx| -> Result<_> {
+                if let Some(transaction_id) = tx.get::<tables::TxHashNumber>(tx_hash)? {
+                    if let Some(transaction) = tx.get::<tables::Transactions>(transaction_id)? {
+                        let mut transaction_cursor =
+                            tx.cursor_read::<tables::TransactionBlock>()?;
+                        if let Some(block_number) =
+                            transaction_cursor.seek(transaction_id).map(|b| b.map(|(_, bn)| bn))?
+                        {
+                            if let Some(block_hash) =
+                                tx.get::<tables::CanonicalHeaders>(block_number)?
+                            {
+                                if let Some(block_body) =
```

Comment:
I wonder if this would be nicer using `Option::and_then` and `Option::map`?

Reply:
I tried this first, total mess